pkg/reader/generic/sql.go

```diff
@@ -80,6 +98,19 @@ func (s *SqlReader) ReadTable(tableName string, rowChan chan<- database.Row, opt
 	return s.PublishRows(rows, rowChan)
 }
 
+// BuildQuery builds the query that will be used to read the table
+func (s *SqlReader) BuildQuery(tableName string, opts reader.ReadTableOpt) (sq.SelectBuilder, error) {
+	var query sq.SelectBuilder
+
+	query = sq.Select(opts.Columns...).From(s.QuoteIdentifier(tableName))
+
+	if opts.Limit > 0 {
```

Comment:
If there is not limit/constraint do we need the sub query?

Reply:
The subquery is only applied if you have relationships, the Limit is applied if you specified a limit in the config file.

Reply:
Yes I know but if the relationship has not constaint (aka limit or where) then it can just be a normal join right?

Reply:
Ah, yes it will be a normal Join if you don't set any limit/constraints, this is the default behavior already.